src/main/java/graphql/execution/SimpleDataFetcherExceptionHandler.java

```diff
@@ -0,0 +1,25 @@
+package graphql.execution;
+
+import graphql.ExceptionWhileDataFetching;
+import graphql.language.SourceLocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The standard handling of data fetcher error involves placing a {@link ExceptionWhileDataFetching} error
+ * into the error collection
+ */
+public class SimpleDataFetcherExceptionHandler implements DataFetcherExceptionHandler {
+    private static final Logger log = LoggerFactory.getLogger(SimpleDataFetcherExceptionHandler.class);
+
+    @Override
+    public void accept(DataFetcherExceptionHandlerParameters handlerParameters) {
+        Exception exception = handlerParameters.getException();
+        SourceLocation sourceLocation = handlerParameters.getField().getSourceLocation();
+        ExecutionPath path = handlerParameters.getPath();
+
+        ExceptionWhileDataFetching error = new ExceptionWhileDataFetching(path, exception, sourceLocation);
+        handlerParameters.getExecutionContext().addError(error);
+        log.warn(error.getMessage(), exception);
```

Comment:
In my opinion, end users should be responsible for defining the data fetching exception logging. If a client wants simple log-everything behavior, they can easily:
```
SimpleDataFetcherExceptionHandlerWithLogging extends SimpleDataFetcherExceptionHandler {

    @Override
    public void accept(DataFetcherExceptionHandlerParameters params) {
        super(params)
        log.warn(...)
```

In our API, we have some commonly thrown exceptions (e.g. expired auth) which we don't want to be logged as warnings. Otherwise, we will send tons of non-actionable events through our logging and alerting infrastructure. To implement my own selective logging logic, I had to disable logging on the entire `ExecutionStrategy` class -- which means I may miss important warnings from it in the future.